src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts

```diff
@@ -0,0 +1,323 @@
+      courseHierarchy: results[0].result.content,
+      enrolledBatchDetails: results[1],
+    })));
+  }
+
+  setActiveContent(selectedContent: string, isSingleContent?: boolean) {
+    if (_.get(this.courseHierarchy, 'children')) {
+      const flattenDeepContents = this.flattenDeep(this.courseHierarchy.children);
+
+      if (selectedContent) {
+        this.activeContent = flattenDeepContents.find(content => content.identifier === selectedContent);
+      } else {
+        this.activeContent = this.firstNonCollectionContent(flattenDeepContents);
+      }
+
+
+      if (this.activeContent) {
+        this.isContentPresent = true;
+        this.initPlayer(_.get(this.activeContent, 'identifier'));
+      }
+    } else if (isSingleContent) {
+      this.activeContent = this.courseHierarchy;
+      this.initPlayer(_.get(this.activeContent, 'identifier'));
+    }
+    this.getContentState();
+  }
+
+  private firstNonCollectionContent(contents) {
+    return contents.find((content) => content.mimeType !== 'application/vnd.ekstep.content-collection');
+  }
+
+  private flattenDeep(contents) {
+    if (contents) {
+      return contents.reduce((acc, val) => {
+        if (val.children) {
+          acc.push(val);
+          return acc.concat(this.flattenDeep(val.children));
+        } else {
+          return acc.concat(val);
+        }
+      }, []);
+    }
+  }
+
+  private initPlayer(id: string) {
+    const options: any = { courseId: this.collectionId };
+
+    if (this.batchId) {
+      options.batchId = this.batchId;
+    }
+    this.courseConsumptionService.getConfigByContent(id, options)
+      .pipe(first(), takeUntil(this.unsubscribe))
+      .subscribe(config => {
+        this.playerConfig = config;
+        this.showLoader = false;
+      }, (err) => {
+        this.showLoader = false;
+        this.toasterService.error(this.resourceService.messages.stmsg.m0009);
+      });
+  }
+
+  onTocCardClick(event: any) {
+    if (_.get(event, 'data')) {
+      this.activeContent = event.data;
+      this.initPlayer(_.get(this.activeContent, 'identifier'));
+    }
+  }
+
+  private getContentState() {
+    const req = {
+      userId: this.userService.userid,
+      courseId: this.courseId,
+      contentIds: this.parseChildContent(),
+      batchId: this.batchId
+    };
+    this.courseConsumptionService.getContentState(req)
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(res => {
+        this.contentStatus = res.content || [];
+        this.calculateProgress();
+      }, err => console.error(err, 'content read api failed'));
+  }
+
+  private parseChildContent() {
+    const contentIds = [this.activeContent.identifier];
+    const model = new TreeModel();
+    const mimeTypeCount = {};
+    const treeModel: any = model.parse(this.courseHierarchy);
+    treeModel.walk((node) => {
+      if (node.model.mimeType !== 'application/vnd.ekstep.content-collection') {
+        if (mimeTypeCount[node.model.mimeType]) {
```

Comment:
change this to 
`
mimeTypeCount[node.model.mimeType] = mimeTypeCount[node.model.mimeType]+1 || 1;
`